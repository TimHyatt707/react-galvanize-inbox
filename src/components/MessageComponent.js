import React from 'react';
var classNames = require('classnames');

export default function MessageComponent({ selected, message }) {
  let star;
  let labels = message.labels.map(i => {
    return (
      <span className="label label-warning">
        {i}
      </span> || ''
    );
  });
  let classes = classNames({
    row: true,
    message: true,
    read: message.read,
    selected: selected
  });
  if (message.starred) {
    star = classNames({
      star: true,
      fa: true,
      'fa-star': true
    });
  } else {
    star = classNames({
      star: true,
      fa: true,
      'fa-star-o': true
    });
  }
  let element = (
    <div className={classes}>
      <div className="col-xs-1">
        <div className="row">
          <div className="col-xs-2">
            <input type="checkbox" checked="checked" />
          </div>
          <div className="col-xs-2">
            <i className={star} />
          </div>
        </div>
      </div>
      <div className="col-xs-11">
        {labels}
        {message.subject}
      </div>
    </div>
  );
  return element;
}
