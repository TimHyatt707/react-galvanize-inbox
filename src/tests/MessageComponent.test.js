import React from 'react';
import { shallow, mount } from 'enzyme';

import MessageComponent from './../components/MessageComponent.js';

let selected = true;

let checked = true;

let message = {
  id: 3,
  subject:
    'Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!',
  read: false,
  starred: true,
  labels: ['dev']
};

let message2 = {
  id: 3,
  subject:
    'Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!',
  read: true,
  starred: false,
  labels: ['dev']
};

const shallowWrapper = shallow(
  <MessageComponent selected={selected} checked={checked} message={message} />
);

describe('<MessageComponent/>', () => {
  const onMarkReadMessage = jest.fn();

  const onSelectMessage = jest.fn();

  const onDeselectMessage = jest.fn();

  const onStarMessage = jest.fn();

  const onUnstarMessage = jest.fn();
  it('snapshot', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
  it('subject line should render', () => {
    expect(shallowWrapper.find('div.col-xs-11').childAt(0)).toHaveLength(1);
  });
  it('read CSS class is present when read is true', () => {
    expect(
      mount(<MessageComponent message={message2} />).find('.read')
    ).toBeTruthy();
  });
  it('unread CSS class is present when read is false', () => {
    expect(
      mount(<MessageComponent message={message} />).find('.unread')
    ).toBeTruthy();
  });
  it('read CSS class is present starrred', () => {
    expect(
      mount(<MessageComponent message={message} />).find('i.star')
    ).toBeTruthy();
  });
  it('read CSS class is present selected', () => {
    expect(
      mount(<MessageComponent message={message} selected={true} />).find(
        'div.selected'
      )
    ).toBeTruthy();
  });
  it('read CSS class is present selected', () => {
    expect(
      mount(<MessageComponent message={message} selected={true} />).find(
        'div.checked'
      )
    ).toBeTruthy();
  });
  it('onMarkAsReadMessage callback is fired when clicked', () => {
    mount(
      <MessageComponent
        message={message}
        selected={true}
        onMarkReadMessage={onMarkReadMessage}
      />
    )
      .find('div.col-xs-11')
      .simulate('click');
    expect(onMarkReadMessage).toBeCalled();
  });
  //
  it('onSelectMessage callback is fired when clicked', () => {
    mount(
      <MessageComponent
        message={message}
        selected={false}
        checked={false}
        onSelectMessage={onSelectMessage}
        onDeselectMessage={onDeselectMessage}
      />
    )
      .find('input')
      .simulate('click');
    expect(onDeselectMessage).toBeCalled();
  });
  //
  it('onDeselectMessage callback is fired when clicked', () => {
    mount(
      <MessageComponent
        message={message}
        selected={true}
        onSelectMessage={onSelectMessage}
        onDeselectMessage={onDeselectMessage}
      />
    )
      .find('input')
      .simulate('click');
    expect(onDeselectMessage).toBeCalled();
  });
  //
  it('onStarMessage callback is fired when clicked', () => {
    mount(<MessageComponent message={message2} onStarMessage={onStarMessage} />)
      .find('i')
      .simulate('click');
    expect(onStarMessage).toBeCalled();
  });
  //
  it('onUnstarMessage callback is fired when clicked', () => {
    mount(
      <MessageComponent message={message} onUnstarMessage={onUnstarMessage} />
    )
      .find('i')
      .simulate('click');
    expect(onUnstarMessage).toBeCalled();
  });
});
