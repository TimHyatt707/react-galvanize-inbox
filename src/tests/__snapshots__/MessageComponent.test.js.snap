// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageComponent/> snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="row message unread selected"
  >
    <div
      className="col-xs-1"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-2"
        >
          <input
            checked={true}
            onClick={[Function]}
            type="checkbox"
          />
        </div>
        <div
          className="col-xs-2"
        >
          <i
            className="star fa fa-star"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="col-xs-11"
      onClick={[Function]}
    >
      <span
        className="label label-warning"
      >
        dev
      </span>
      Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!
    </div>
  </div>,
  "nodes": Array [
    <div
      className="row message unread selected"
    >
      <div
        className="col-xs-1"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-2"
          >
            <input
              checked={true}
              onClick={[Function]}
              type="checkbox"
            />
          </div>
          <div
            className="col-xs-2"
          >
            <i
              className="star fa fa-star"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="col-xs-11"
        onClick={[Function]}
      >
        <span
          className="label label-warning"
        >
          dev
        </span>
        Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <MessageComponent
        checked={true}
        message={
          Object {
            "id": 3,
            "labels": Array [
              "dev",
            ],
            "read": false,
            "starred": true,
            "subject": "Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!",
          }
        }
        selected={true}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "checked": true,
          "message": Object {
            "id": 3,
            "labels": Array [
              "dev",
            ],
            "read": false,
            "starred": true,
            "subject": "Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!",
          },
          "selected": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="row message unread selected"
        >
          <div
            className="col-xs-1"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-2"
              >
                <input
                  checked={true}
                  onClick={[Function]}
                  type="checkbox"
                />
              </div>
              <div
                className="col-xs-2"
              >
                <i
                  className="star fa fa-star"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="col-xs-11"
            onClick={[Function]}
          >
            <span
              className="label label-warning"
            >
              dev
            </span>
            Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="row message unread selected"
        >
          <div
            className="col-xs-1"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-2"
              >
                <input
                  checked={true}
                  onClick={[Function]}
                  type="checkbox"
                />
              </div>
              <div
                className="col-xs-2"
              >
                <i
                  className="star fa fa-star"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="col-xs-11"
            onClick={[Function]}
          >
            <span
              className="label label-warning"
            >
              dev
            </span>
            Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MessageComponent
    checked={true}
    message={
      Object {
        "id": 3,
        "labels": Array [
          "dev",
        ],
        "read": false,
        "starred": true,
        "subject": "Use the 1080p HTTP feed, then you can parse the cross-platform hard drive!",
      }
    }
    selected={true}
  />,
}
`;
